core/src/main/java/com/sequenceiq/cloudbreak/service/rdsconfig/DbUsernameConverterService.java

```diff
@@ -0,0 +1,47 @@
+package com.sequenceiq.cloudbreak.service.rdsconfig;
+
+import java.util.Locale;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+@Service
+public class DbUsernameConverterService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DbUsernameConverterService.class);
+
+    public String toConnectionUsername(String host, String dbUser) {
+        if (DbHostnameMatcher.isAzure(host)) {
+            LOGGER.debug("Detected Azure database server {}, appending short hostname to connection username", host);
+            return dbUser + "@" + host.substring(0, host.indexOf('.'));
+        }
+
+        return dbUser;
+    }
+
+    /**
+     * Converts a connection username to a db username.
+     * On azure, connection username follows the 'username@short-hostname' pattern.
+     * On postgres, however, users have to be in 'username' format, without the '@' sign => code has to take username.
+     *
+     * @param username a connection username
+     * @return returns the username part before the '@' sign
+     */
+    public String toDatabaseUsername(String username) {
+        int atIndex = username.indexOf("@");
+        return atIndex > -1 ? username.substring(0, atIndex) : username;
+    }
+
+    private static class DbHostnameMatcher {
```

Comment:
why is this separate inner class necessary?

Reply:
The goal was to express the higher cohesion between the isAzure method and the DbHostnameMatcher that is used from one place (hence embedded class). If you think this decreases readability, I can revert it back to a private method and a constant within DbUsernameConverterService.

Reply:
As a result of our private discussion I got rid of the inner class.